lms/envs/common.py

```diff
@@ -1610,7 +1610,20 @@
 # and defines some "args" function that returns a list with the command arguments to execute.
 REQUIRE_ENVIRONMENT = "node"
 
-
+# In production, the Django pipeline appends a file hash to JavaScript file names.
+# This makes it difficult for RequireJS to load its requirements, since module names
+# specified in JavaScript code do not include the hash.
+# For this reason, we calculate the actual path including the hash on the server
+# when rendering the page.  We then override the default paths provided to RequireJS
+# so it can resolve the module name to the correct URL.
+#
+# If you want to load JavaScript dependencies using RequireJS
+# but you don't want to include those dependencies in the JS bundle for the page,
+# then you need to add the js urls in this list.
+REQUIRE_JS_PATH_OVERRIDES = [
```

Comment:
@muhammad-ammar - @clrux is looking into using this mechanism to allow an XBlock to load a client-side dependency with the correct MD5 hash. I hadn't realized that this was an LMS-only feature currently. I think we need to generalize this so that this is implemented in common, and then LMS and Studio can add their unique URLs to the list. FYI @cahrens.